themes/DeepWave.clinkprompt

```diff
@@ -0,0 +1,499 @@
+	return sgr_fg(fg)..get_dts()..sgr_fg()
+end
+
+-- Build the Left Side Two-line Prompt 2nd-Line Prompt
+local function build_prompt_left_line2(muted)
+	local str = ""
+	local line2_fg = theme_defns["line2"].fg
+	-- If we were asked for a Muted Prompt, use a Muted Colour instead
+	if (muted) then
+		line2_fg = theme_defns["line2"].muted_fg
+	end
+	-- 2nd-Line Segment
+	return sgr_fg()..sgr_fg(line2_fg)..ucs2to8(def_sep_nobg)..sgr_fg().." "
+end
+
+--------------------------------------------------------------------------------
+-- clinkprompt Filter Integration
+
+-- Main Prompt
+local dw_prompt = clink.promptfilter(5)
+
+-- Set the Prompt
+function dw_prompt:filter()
+	if (dw.enable_twoline) then
+		-- In a Two-line Prompt, the Right Prompt needs to be on the same line as the Left Prompt, so we put it here instead of in :rightfilter()
+		local left = build_prompt_left(false)
+		local right = ""
+		local pad = ""
+		if (dw.enable_timestamp) then
+			-- Show the Timestamp on the Right
+			right = build_prompt_right(false)
+			local left_len = console.cellcount(left)
+		    local right_len = console.cellcount(right)
+			if left_len + 4 + right_len >= console.getwidth() then
+				right = ""
+			else
+				pad = string.rep(" ", console.getwidth() - (left_len + right_len))
+			end
+		end
+		return left..pad..right.."\n"..build_prompt_left_line2(false)
+	else
+		return build_prompt_left(false)
+	end
+end
+
+-- Set the Prompt (Right)
+function dw_prompt:rightfilter()
+	-- In a Two-line Prompt, the Right Prompt needs to be on the same line as the Left Prompt, so we put it in :filter() instead of here
+	if ((dw.enable_timestamp) and (not dw.enable_twoline)) then
+		return build_prompt_right(false)
+	else
+		return nil
+	end
+end
+
+-- Set the (Muted) Transient Prompt
+function dw_prompt:transientfilter()
+	if (dw.enable_twoline) then
+		-- In a Two-line Prompt, we don't show the Left Prompt anymore
+		return build_prompt_left_line2(true)
+	else
+		return build_prompt_left(true)
+	end
+end
+
+-- Set the (Muted) Transient Prompt (Right)
+function dw_prompt:transientrightfilter()
+	-- In a Two-line Prompt, we don't show the Previous Right Prompts anymore
+	if ((dw.enable_timestamp) and (not dw.enable_twoline)) then
+		-- 20241018: If the existing prompt string is _EXACTLY_ the same as the updated one, then nothing is shown
+		-- DEBUG Is this a bug?
+		return sgr_fg()..build_prompt_right(true, dw.enable_twoline)
+	else
+		return nil
+	end
+end
+
+--------------------------------------------------------------------------------
+-- clinkprompt Event Handlers
+
+-- Command Line Header
+local function onbeginedit()
+	set_settings()
+end
+
+-- Register our Event Handlers
+clink.onbeginedit(onbeginedit)
+
+--------------------------------------------------------------------------------
+-- clinkprompt Module Handlers
+
+-- Upon Prompt Show
+-- 20241018: This does not seem to be executed upon 'clink config prompt show', so we call it from ondeactivate() instead
```

Comment:
Hm, I'll try it later today.  Last I had checked the demo callback was being called, but maybe it got broken at some point.

Reply:
Yeah, if you try the sample.clinkprompt it doesn't work either.. I'm not sure what is supposed to happen though..

Reply:
Yup, I broke it before the official release, as a side effect of some cleanup changes. 

I'll publish a fix in a week or two.

In the meantime I wouldn't recommend using `ondeactivate`, because that gets called whenever you use `clink config prompt use` to switch to a different clinkprompt.  Users will not expect to see a demo of this prompt shown whenever they select some other prompt.

Tonight I'll try to see if there's another way to detect that `clink config prompt show` is happening.

Reply:
> In the meantime I wouldn't recommend using `ondeactivate`, because that gets called whenever you use `clink config prompt use` to switch to a different clinkprompt. Users will not expect to see a demo of this prompt shown whenever they select some other prompt.

Yeah, I agree.. that should be a demo thing only... So this ought to be revised... Activate and Deactivate should be used for only exactly that. 

> Tonight I'll try to see if there's another way to detect that `clink config prompt show` is happening.

Actually, I like what `clink config prompt show` is doing now.. It gives users a brief preview of the prompt before switching to it. 

But perhaps there can be a `clink config prompt info` or something instead of demo.. that just queries the prompt for information or status/settings/etc. What do you think?

Reply:
(Where it gets a little more involved is if the prompt has a `:rightfilter` function, because then the author has to do the math/etc for right aligning the right side prompt demo.  And that's why I'm considering making a way for demo functions to essentially say "ok Clink, do your thing and print a prompt here before/after I print other things".)

Reply:
Commit d1f34d7f0633033fdf29fcc94fe1133c37b869d1 fixes the `demo` callback.

Reply:
> Only, I don't want to make it something that prompt authors feel they need to add. And I'm not sure what "automatic" info would be meaningful to generate automatically for prompts that don't implement their own custom info.
> 
> What I've been thinking about is a way for a prompt's demo function to let Clink automatically show the demo, and then the demo function could just add some custom info, like you wanted. To minimize work the prompt author would need to do.
> 
> I think `clink config prompt show` is a reasonable place to hook that into.

Hmm, I understand... I liked that there was automatically a nice comparison between the old prompt and the new, and this was an apples to apples comparison. Same path, default settings etc etc. Like, this is what you have currently, and given everything the same, this is what it would look like.

But okie, that's workable too, to provide more flexibility to prompt authors to maybe showcase more different settings and options with a `demo()` if they wished... ^_^

I've been assuming the basic prompt from `render()` would automatically/always be there in `prompt show` .. but now I see that's different, `demo()` does everything. So give me a day or two to think about what I want to show.. I mean demonstrate.. ha ha!

Anyway, truthfully, I'm really supposed to be working on other things right now, but this little prompt thing has been a fun (I mean naughty) distraction for me. Ha ha .. Thank you very much for clink and the distraction.. ^_^